packages/xo-web/src/xo-app/settings/acls/index.js

```diff
@@ -80,92 +80,62 @@ const ACL_ACTIONS = [
-      const { xoObjects } = this.props
-      const { acls, roles } = this.state
-      const resolvedAcls = filter(
-        map(acls, ({ id, subject, object, action }) => ({
-          id,
-          subject: subjects[subject] || subject,
-          object: xoObjects[object] || object,
-          action: roles[action] || action,
-        })),
-        ({ subject, object, action }) =>
-          subject && object && action && object.type !== 'VM-snapshot'
-      )
-      this.setState({
-        resolvedAcls,
-      })
-    }
-
-    const unsubscribeAcls = subscribeAcls(acls =>
-      this.setState({ acls }, refresh)
-    )
-    const unsubscribeRoles = subscribeRoles(roles =>
-      this.setState({ roles: keyBy(roles, 'id') }, refresh)
-    )
-    const unsubscribeGroups = subscribeGroups(groups => {
-      groups = keyBy(groups, 'id')
-      refresh({
-        ...pickBy(subjects, subject => subject.type === 'user'),
-        ...groups,
-      })
-    })
-    const unsubscribeUsers = subscribeUsers(users => {
-      users = keyBy(users, 'id')
-      refresh({
-        ...pickBy(subjects, subject => subject.type === 'group'),
-        ...users,
+const AclTable = [
+  connectStore(() => ({
+    xoObjects: createSelector(
+      createGetObjectsOfType('host'),
+      createGetObjectsOfType('network'),
+      createGetObjectsOfType('pool'),
+      createGetObjectsOfType('SR'),
+      createGetObjectsOfType('VM'),
+      createGetObjectsOfType('VM-snapshot'),
+      (hosts, networks, pools, srs, vms, snapshots) => ({
+        ...hosts,
+        ...networks,
+        ...pools,
+        ...snapshots,
+        ...srs,
+        ...vms,
       })
-    })
-
-    this.componentWillUnmount = () => {
-      unsubscribeAcls()
-      unsubscribeGroups()
-      unsubscribeRoles()
-      unsubscribeUsers()
-    }
-  }
-
-  render () {
-    const { resolvedAcls = [] } = this.state
-
-    return isEmpty(resolvedAcls) ? (
-      <p>
-        <em>{_('aclNoneFound')}</em>
-      </p>
-    ) : (
-      <SortedTable
-        actions={ACL_ACTIONS}
-        collection={resolvedAcls}
-        columns={ACL_COLUMNS}
-      />
-    )
-  }
-}
+    ),
+  })),
+  addSubscriptions({
+    acls: subscribeAcls,
+    roles: subscribeRoles,
+    groups: subscribeGroups,
+    users: subscribeUsers,
+  }),
+  provideState({
+    initialState: () => ({
+      acls: [],
+    }),
+    computed: {
+      acls: ({ roles, subjects }, { xoObjects, acls }) =>
+        filter(
+          map(acls, ({ id, subject, object, action }) => ({
+            id,
+            subject: subjects[subject],
+            object: xoObjects[object],
+            action: roles[action],
+          })),
+          ({ subject, object, action }) =>
+            subject && object && action && object.type !== 'VM-snapshot'
+        ),
+      groups: (_, { groups }) => keyBy(groups, 'id'),
+      roles: (_, { roles }) => keyBy(roles, 'id'),
+      subjects: ({ users, groups }) => ({ ...users, ...groups }),
```

Comment:
IMHO, is not necessary simply test if the subject is in the groups collection or the users collections.